main.tf

```diff
@@ -115,7 +115,7 @@ resource "aws_eip" "default" {
 module "ansible" {
   source    = "git::https://github.com/cloudposse/tf_ansible.git?ref=tags/0.3.4"
   arguments = "${var.ansible_arguments}"
-  envs      = "${compact(concat(var.ansible_envs, list("host=${var.associate_public_ip_address ? aws_eip.default.public_ip : aws_instance.default.private_dns }")))}"
```

Comment:
this won't work unless your provisioning inside of EC2. It's better to leverage the `public_dns` which will work in both scenarios.

Reply:
I will concede on this point. It's bad practice to open up SSH to the world on all servers, so this is not a use-case to optimize for.  We'll require private access and consider adding support for jumpbox later.